File: tempk.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>


int main(void)
{

    string input_type = get_string("what type of degree do you want to convert? celcius, farenheit or kelvin ?\n");
    string output_type = get_string("to what should it be converted?\n");

    if (strcmp(input_type, "c") == 0 || strcmp(input_type, "C") == 0 || strcmp(input_type, "celsius") == 0)
    {
        if (strcmp(output_type, "f") == 0 || strcmp(output_type, "F") == 0 || strcmp(output_type, "fahrenheit") == 0)
        {
          float degree = get_float("how many degrees?\n");
          printf("the temprature you gave me results to %f degrees!\n", degree * 9/5 + 32);
        }

         else if (strcmp(output_type, "k") == 0 || strcmp(output_type, "K") == 0 || strcmp(output_type, "kelvin") == 0)
        {
           float degree = get_float("how many degrees?\n");
           printf("the temprature you gave me results to %f degrees!\n", degree + 273.15 );
        }
        else
        {
            printf("unit not supported\n");
        }
        }


    if (strcmp(input_type, "f") == 0 || strcmp(input_type, "F") == 0 || strcmp(input_type, "fahrenheit") == 0)

    {
        if (strcmp(output_type, "c") == 0 || strcmp(output_type, "C") == 0 || strcmp(output_type, "celsius") == 0)
        {
            float degree = get_float("how many degrees?\n");
            printf("the temprature you gave me results to %f degrees!\n", (degree - 32) * 5/9);
        }
          else if (strcmp(output_type, "k") == 0 || strcmp(output_type, "K") == 0 || strcmp(output_type, "kelvin") == 0)
        {
           float degree = get_float("how many degrees?\n");
           printf("the temprature you gave me results to %f degrees!\n", (degree - 32) * 5/9 + 273.15 );

         }
         else
        {
            printf("unit not supported\n");
        }

    }

      if (strcmp(input_type, "k") == 0 || strcmp(input_type, "K") == 0 || strcmp(input_type, "kelvin") == 0)
      {
        if (strcmp(output_type, "c") == 0 || strcmp(output_type, "C") == 0 || strcmp(output_type, "celsius") == 0)
        {
            float degree = get_float("how many degrees?\n");
            printf("the temprature you gave me results to %f degrees!\n", degree - 273.15 );
        }

         if (strcmp(output_type, "f") == 0 || strcmp(output_type, "F") == 0 || strcmp(output_type, "fahrenheit") == 0)
        {
          float degree = get_float("how many degrees?\n");
          printf("the temprature you gave me results to %f degrees!\n", 1.8 * (degree - 273.15) + 32 );

      }
       else
        {
            printf("unit not supported\n");
        }

      }
}

File: tempk.py
print("This is a temperature converter")
print("It converts Celsius to Fahrenheit and the opposite")

temp_unit = input("What temperature unit do you want to convert? \nType 1 for Celsius, 2 for Fahrenheit, or 3 for Kelvin: ")
temp_unit = float(temp_unit)

temp_convert = input("What should it be converted to? \nType 1 for Celsius, 2 for Fahrenheit, or 3 for Kelvin: ")
temp_convert = float(temp_convert)

degree = input("And how many degrees do you want to convert? ")
degree = float(degree)

if temp_unit == 1:
    if temp_convert == 2:
        degree_outcome = (degree * 9/5 + 32)
        print(f"The result is {degree_outcome} Fahrenheit")
    elif temp_convert == 3:
        degree_outcome = (degree + 273.15)
        print(f"The result is {degree_outcome} Kelvin")
    else:
        print("The convert unit is not correct or does not exist")

elif temp_unit == 2:
    if temp_convert == 1:
        degree_outcome = ((degree - 32) * 5/9)
        print(f"The result is {degree_outcome} Celsius")
    elif temp_convert == 3:
        degree_outcome = ((degree - 32) * 5/9 + 273.15)
        print(f"The result is {degree_outcome} Kelvin")
    else:
        print("The convert unit is not correct or does not exist")

elif temp_unit == 3:
    if temp_convert == 1:
        degree_outcome = (degree - 273.15)
        print(f"The result is {degree_outcome} Celsius")
    elif temp_convert == 2:
        degree_outcome = ((degree - 273.15) * 9/5 + 32)
        print(f"The result is {degree_outcome} Fahrenheit")
    else:
        print("The convert unit is not correct or does not exist")

else:
    print("This unit is not supported")

